src/button.rs

```diff
@@ -1,43 +1,29 @@
 use core::convert::Infallible;
-use stm32f4xx_hal as hal;
-
 use embedded_hal::digital::v2::InputPin;
-use hal::timer::{Instant, MonoTimer};
 
 pub struct Button<T: InputPin> {
     pin: Debouncer<T>,
-    timer: MonoTimer,
     button_state: ButtonState,
-    long_press_timeout_ticks: u32,
 }
 
 pub enum ButtonEvent {
     /// The button has just been pressed down.
-    Pressed,
-
-    /// The button was released before the "long press" timeout.
-    ShortRelease,
+    Press,
 
-    /// The button has been held for at least the "long press" timeout.
-    LongPress,
-
-    /// The button has been released after a "long press".
-    LongRelease,
+    /// The button was released.
+    Release,
 }
 
 enum ButtonState {
     Released,
-    Pressed(Instant),
-    LongPressed,
+    Pressed,
```

Comment:
super nit: reverse the order of `Pressed` and `Released`?

Reply:
I left it as the first one because that is its initial state.